client/components/registration/registration.js

```diff
@@ -1034,4 +1039,7 @@ Registration.propTypes = {
     content: PropTypes.object,
   }).isRequired,
   authenticate: PropTypes.func.isRequired,
+  setUserData: PropTypes.func.isRequired,
+  userData: PropTypes.object.isRequired,
```

Comment:
pay attention, I removed `userData` and `setUserData` from this component, I centralized all the operations in the status component to simplify and avoid duplication, therefore these lines should not be added back.